ql/src/java/org/apache/hadoop/hive/ql/udf/ptf/ValueBoundaryScanner.java

```diff
@@ -768,6 +774,9 @@ public static SingleValueBoundaryScanner getBoundaryScanner(BoundaryDef start, B
     case "string":
       return new StringPrimitiveValueBoundaryScanner(start, end, exprDef, nullsLast);
     default:
+      if (typeString.startsWith("char") || typeString.startsWith("varchar")) {
```

Comment:
putting this into default looks strange to me, why not handle similarly to decimal as above

Reply:
can't do that. the entries aren't fixed char or varchar, they are like char(10) or char(5) or varchar(5) or varchar(6) like that. So putting char or varchar in switch-case won't work

Reply:
Done. As discussed I pulled all of them together to avoid multiple branches.